File: php/Sully.php
#!/usr/bin/php
<?PHP
define('C', '#!/usr/bin/php%c<?PHP%cdefine(%cC%c, %c%s%c);%c%c%ci = %d;%cif (%cargv[0][5] != %c.%c)%c	%ci--;%c%cn = sprintf(%cSully_%%d.php%c, %ci);%c%cf = fopen(%cn, %cw+%c);%cfprintf(%cf, C, 10, 10, 39, 39, 39, C, 39, 10, 10, 36, %ci, 10, 36, 39, 39, 10, 36, 10, 36, 34, 34, 36, 10, 36, 36, 34, 34, 10, 36, 36, 10, 36, 10, 36, 34, 34, 36, 10, 36, 10, 36, 10);%cfclose(%cf);%c%cn = sprintf(%cphp Sully_%%d.php%c, %ci);%cif (%ci > 0)%c	system(%cn);%c?>');

$i = 5;
if ($argv[0][5] != '.')
	$i--;
$n = sprintf("Sully_%d.php", $i);
$f = fopen($n, "w+");
fprintf($f, C, 10, 10, 39, 39, 39, C, 39, 10, 10, 36, $i, 10, 36, 39, 39, 10, 36, 10, 36, 34, 34, 36, 10, 36, 36, 34, 34, 10, 36, 36, 10, 36, 10, 36, 34, 34, 36, 10, 36, 10, 36, 10);
fclose($f);
$n = sprintf("php Sully_%d.php", $i);
if ($i > 0)
	system($n);
?>

File: Sully.c
#include <stdio.h>
#include <stdlib.h>
#define C "#include <stdio.h>%1$c#include <stdlib.h>%1$c#define C %2$c%3$s%2$c%1$c%1$cint	main(int ac, char %4$c%4$cav){%1$c	int i = %5$d;%1$c	char n[49];%1$c	FILE%4$c f = NULL;%1$c%1$c	if (ac && av[0][7] != %6$c%7$c0%6$c)%1$c		i--;%1$c	sprintf(n, %2$cSully_%%d.c%2$c, i);%1$c	f = fopen(n, %2$cw+%2$c);%1$c	fprintf(f, C, 10, 34, C, 42, i, 39, 92);%1$c	fclose(f);%1$c	sprintf(n, %2$cgcc -Wall -Wextra -Werror -o Sully_%%1$d Sully_%%1$d.c%2$c, i);%1$c	system(n);%1$c	sprintf(n, %2$c./Sully_%%d%2$c, i);%1$c	if (i > 0)%1$c		system(n);%1$c}"

int	main(int ac, char **av){
	int i = 5;
	char n[49];
	FILE* f = NULL;

	if (ac && av[0][7] != '\0')
		i--;
	sprintf(n, "Sully_%d.c", i);
	f = fopen(n, "w+");
	fprintf(f, C, 10, 34, C, 42, i, 39, 92);
	fclose(f);
	sprintf(n, "gcc -Wall -Wextra -Werror -o Sully_%1$d Sully_%1$d.c", i);
	system(n);
	sprintf(n, "./Sully_%d", i);
	if (i > 0)
		system(n);
}